File: ordenacao/selection_sort.js
const selection_sort = (nums) => {
	const N = nums.length;
	for(let i = 0; i < N - 1; i++){
		let minIndex = i;
		for(let j = i + 1; j < N; j++){
			if(nums[j] < nums[minIndex])
				minIndex = j;
		if (minIndex != i)
			swap(nums, i, minIndex);
		}
	}
	return nums;
};

const swap = (nums, a, b) => {
	const aux = nums[a];
	nums[a] = nums[b];
	nums[b] = aux;
};

console.log(selection_sort([20, 9, 86, -2, 16]));

File: src/selectionSort/SelectionSort.java
package selectionSort;

public class SelectionSort {
    public static void main(String[] args) {
        int[] valores = {34, 78, 9, 2, 34, 1, 11, 4, 89, -1};
        System.out.println("Array antes da ordenação: ");
        for (int valor : valores) {
            System.out.print(valor + " ");
        }
        System.out.println("\n-----------------------------");
        System.out.println("Array após a ordenação: ");
        selectionSort(valores);
        for (int valor : valores) {
            System.out.print(valor + " ");
        }
    }

    public static void selectionSort(int[] valores){
        for (int i = 0; i < valores.length; i++){

            int valor_menor = i;
            for(int j = i + 1; j < valores.length; j++){
                if (valores[j] < valores[valor_menor])
                    valor_menor = j;
            }

            int aux = valores[i];
            valores[i] = valores[valor_menor];
            valores[valor_menor] = aux;
        }
    }
}
